code/controllers/subsystem/ships.dm

```diff
@@ -78,9 +78,9 @@ var/global/list/ftl_weapons_consoles = list()
 			return faction_A.relations[i]
 
 
-/datum/subsystem/ship/proc/calculate_damage_effects(var/datum/starship/S)
+/datum/subsystem/ship/proc/calculate_damage_effects(var/datum/starship/S, var/datum/component/weapon/W)
```

Comment:
Leave this be, and add a for loop below the calculate damage effects for all weapons on the ship.